source/common/network/dns_impl.cc

```diff
@@ -135,5 +160,13 @@ ActiveDnsQuery* DnsResolverImpl::resolve(const std::string& dns_name, ResolveCb
   }
 }
 
+void DnsResolverImpl::PendingResolution::getHostByName(int family) {
+  ares_gethostbyname(channel_, dns_name_.c_str(),
```

Comment:
As discussed IRL last week, `ares_gethostbyname()` can implement the IPv6-IPv4 fallback with `AF_UNSPEC` family set (but is badly documented). However, I think your current implementation gets us closer to where we want to be in #978, since it will allow for the return of both IPv6 and IPv4 addresses in a list in a future PR, where we can support the caller deciding upon fallback as a result of `connect()` failure. So, I reckon it's good to keep the explicit fallback in this PR.